src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile

```diff
@@ -3,7 +3,7 @@
 # and prepares for execution of gitplugin tests.
 #
 
-FROM ubuntu:22.04
+FROM ubuntu:xenial
```

Comment:
most likely the line 24 is incorrect

`RUN cat /home/git/unsafe.pub >> /home/git/.ssh/authorized_keys`
is likely creating the file using the default keymask  (infact the line 19 `mkdir -p /home/git/.ssh` is probably just as bad.

the `.ssh` folder should be `0700` *and* be owned by the `git` user (not root)

Reply:
+1 with @jtnord . I'm not sure how this Java code works to spin up Docker container (and I'm too scared to try it), but I got the following updated Dockerfile here tested as below (as a first step) and passed through `hadolint` (which has very very good tips and actionnables, like `shellcheck`):

```Dockerfile
FROM ubuntu:22.04

SHELL ["/bin/bash", "-o", "pipefail", "-c"]

# install SSHD, Git and zip, always on the latest available version
# hadolint ignore=DL3008
RUN apt-get update \
    && apt-get install --yes --no-install-recommends \
    ca-certificates \
    openssh-server \
    git \
    zip \
    && rm -rf /var/lib/apt/lists/*

# create a git user and setup its .ssh dir (including public key)
COPY unsafe.pub  /home/git/.ssh/authorized_keys
RUN useradd git -d /home/git \
    && echo "git:git" | chpasswd \
    && mkdir -p /var/run/sshd \
    && chmod 0700 /home/git/.ssh \
    && chmod 0600 /home/git/.ssh/* \
    && chown -R git /home/git

# run SSHD in the foreground with error messages to stderr
ENTRYPOINT ["/usr/sbin/sshd"]
CMD ["-D","-e"]
```

```shell
$ pwd
<my workspace>/acceptance-test-harness/src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer

$ docker build -t sshd ./
# Success

$ docker run --rm --detach --name=sshd-test --publish 2222:22 sshd
# Run in background the container

$ ssh -i ./unsafe -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -p2222 git@localhost whoami
# ..
git
## It works
```